### ISSUE TITLE
Base rule extension: nonblock-statement-body-position

### ISSUE THREAD

**Issue by Memphizzz** [typescript-eslint/typescript-eslint#9320]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+base+rule+extension%22) and found none that match my proposal.
- [X] I have searched the [current extension rule list](https://typescript-eslint.io/rules/?=extension#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the base rule

https://eslint.org/docs/latest/rules/nonblock-statement-body-position

### Description

This rule allows forcing a new line after a condition, e.g.: 

`if (condition) await something();`
vs
`if (condition)
    await something();`

### Fail

```TypeScript
if (condition) await something();
```

### Pass

```TypeScript
if (condition)
    await something();
```

**Comment by Josh-Cena** [comment]
Hi, the base rule is deprecated:

> This rule was deprecated in ESLint v8.53.0. Please use the [corresponding rule](https://eslint.style/rules/js/nonblock-statement-body-position) in [@stylistic/eslint-plugin-js](https://eslint.style/packages/js).

And we have deprecated all formatting rules too: https://typescript-eslint.io/blog/deprecating-formatting-rules to be removed in the next major.
